apps/test/unit/propTypesTest.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import sinon from 'sinon';
+import {expect, assert} from '../util/configuredChai';
+var propTypes = require('@cdo/apps/propTypes');
+
+describe('propTypes module', () => {
+
+  function Foo(){}
+  function Bar(){}
+  function Baz(){}
+
+  describe('whenNoChildOfType prop type', () => {
+    var check;
+    beforeEach(() => {
+      check = ({children, someProp}) => propTypes.whenNoChildOfTypes(Foo)(
+        {children, someProp},
+        'someProp',
+        'SomeComponent'
+      );
+    });
+
+    describe('does not return an error when', () => {
+
+      it('no children are given', () => {
+        expect(check({someProp: 'foo'})).not.to.be.ok;
+      });
+
+      it('no prop is given', () => {
+        expect(check({children: [<Foo />]})).not.to.be.ok;
+      });
+
+    });
+
+    describe('does return an error when', () => {
+
+      it('both a prop and child are given', () => {
+        const error = check({children: [<Foo />], someProp:'foo'});
+        expect(error).to.be.an.instanceOf(Error);
+        expect(error.message).to.equal(
+          'SomeComponent was given a someProp prop and a <Foo> child, ' +
+          'but only one of those is allowed.'
+        );
+      });
+
+    });
+  });
+
+  describe('childrenOfType prop type', () => {
+    var check;
+    beforeEach(() => {
+      check = (...children) => propTypes.childrenOfType(Foo, Bar)(
+        {children},
+        'children',
+        'SomeComponent'
+      );
+    });
+
+    describe('does not return an error when', () => {
+
+      it('no children are given', () => {
+        expect(check()).not.to.be.ok;
```

Comment:
Well, the react documentation doesn't say what it is supposed to return (https://facebook.github.io/react/docs/reusable-components.html), but here is the react code that actually does the check: https://github.com/facebook/react/blob/95ac239cf3c527b70daeb433b7aeeaf9e41be9e5/src/isomorphic/classic/types/checkReactTypeSpec.js#L68.

Looks like it's explicitly doing an instanceof check so a success case presumably is return anything that is not an instance of `Error`. I'll update these tests to reflect that, since at the very least, the existing react implementation behavior is unlikely to change even if it's not documented.